test/integration/controlplane/kube_apiserver_test.go

```diff
@@ -606,3 +626,95 @@ func TestMultiAPIServerNodePortAllocation(t *testing.T) {
+
+	// start apiserver
+	t.Log("starting api server")
+	server := kubeapiservertesting.StartTestServerOrDie(t, instanceOptions, []string{
+		"--endpoint-reconciler-type", "lease",
+		"--advertise-address", fmt.Sprintf("10.0.1.1"),
+	}, etcd)
+	defer server.TearDownFn()
+
+	// verify kube API server has registered and create a client
+	if err := wait.PollImmediate(3*time.Second, 2*time.Minute, func() (bool, error) {
+		client, err := kubernetes.NewForConfig(server.ClientConfig)
+		if err != nil {
+			t.Logf("create client error: %v", err)
+			return false, nil
+		}
+		services, err := client.CoreV1().Services("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+		if err != nil {
+			t.Logf("error fetching services: %v", err)
+			return false, nil
+		}
+		if e, a := []int{server.ServerOpts.SecureServing.BindPort}, getServiceTargetPorts(services); !reflect.DeepEqual(e, a) {
+			t.Logf("error comparing target port in services with bind port set: Bind port %v vs targetPort %v", server.ServerOpts.SecureServing.BindPort, a)
+			return false, nil
+		}
+		endpoints, err := client.CoreV1().Endpoints("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+		if err != nil {
+			t.Logf("error fetching endpoints: %v", err)
+			return false, nil
+		}
+		if e, a := []int{server.ServerOpts.SecureServing.BindPort}, getEndpointPorts(endpoints); !reflect.DeepEqual(e, a) {
+			t.Logf("error comparing target port in endpoints with bind port set: Bind port %v vs targetPort %v", server.ServerOpts.SecureServing.BindPort, a)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("did not find only lease endpoints: %v", err)
+	}
+}
+
+// TestMultiMasterAdvertisePort tests that when using the --advertise-port the targetPort in the kubernetes.default.svc server
+// and also as port in the kubernetes.default.svc endpoints and it's the value specified there
+func TestAPIServerAdvertisePort(t *testing.T) {
+	etcd := framework.SharedEtcd()
+	instanceOptions := kubeapiservertesting.NewDefaultTestServerOptions()
+
+	// start apiserver
+	t.Log("starting api server")
+	server := kubeapiservertesting.StartTestServerOrDie(t, instanceOptions, []string{
+		"--endpoint-reconciler-type", "lease",
+		"--advertise-address", fmt.Sprintf("10.0.1.1"),
+		"--advertise-port", "8443",
+	}, etcd)
+	defer server.TearDownFn()
+
+	// verify kube API server has registered and create a client
+	if err := wait.PollImmediate(3*time.Second, 2*time.Minute, func() (bool, error) {
+		client, err := kubernetes.NewForConfig(server.ClientConfig)
+		if err != nil {
+			t.Logf("create client error: %v", err)
+			return false, nil
+		}
+		services, err := client.CoreV1().Services("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+		if err != nil {
+			t.Logf("error fetching services: %v", err)
+			return false, nil
+		}
+		if e, a := []int{server.ServerOpts.SecureServing.BindPort}, getServiceTargetPorts(services); !reflect.DeepEqual(e, a) {
```

Comment:
in this case both should be different, no? 
the service should have the advertise-port, that is different than the bound port